sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpTransactionCoordinator.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * This provides API to manage AMQP transaction. A transaction is sued where one or more operation in messaging broker
+ * is part of one unit of work. In general a transaction involve with many operations on one message broker entity.
+ *
+ * <p>Distributed Transactions</p>
+ * A distributed transaction spans over different message broker entities.
+ * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#choice-txn-capability-distributed-transactions">Distributed Transactions</a>
+ * @see <a href="https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#section-transactions">Transactions</a>
+ */
+public interface AmqpTransactionCoordinator {
+
+    /**
+     * Completes the transaction. All the work in this transaction will either rollback or committed as one unit of
+     * work.
+     * @param transaction that needs to be completed.
+     * @param isCommit    This flag indicates that the work associated with this transaction should commit or rollback.
+     *
+     * @return a completable {@link Mono}.
+     */
+    Mono<Void> dischargeTransaction(AmqpTransaction transaction, boolean isCommit);
```

Comment:
discharge and declare terms are used by AMQP speceification.

Reply:
`dischargeTransaction` sounds redundant since it already takes a parameter "Transaction". (ie. Similarly, in Event Hubs, we have "send" which takes an EventDataBatch.)

Reply:
changing it to `discharge` and removing `Transaction`